sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/pipeOperators.scala

```diff
@@ -18,26 +18,26 @@
 package org.apache.spark.sql.catalyst.expressions
 
 import org.apache.spark.sql.catalyst.expressions.aggregate.AggregateFunction
-import org.apache.spark.sql.catalyst.trees.TreePattern.{PIPE_OPERATOR_SELECT, RUNTIME_REPLACEABLE, TreePattern}
+import org.apache.spark.sql.catalyst.trees.TreePattern._
 import org.apache.spark.sql.errors.QueryCompilationErrors
 
 /**
- * Represents a SELECT clause when used with the |> SQL pipe operator.
- * We use this to make sure that no aggregate functions exist in the SELECT expressions.
+ * Represents an expression when used with non-aggregate SQL pipe operators like |> SELECT or
+ * |> EXTEND. We use this to make sure that no aggregate functions exist in these expressions.
  */
-case class PipeSelect(child: Expression)
+case class PipeSelect(child: Expression, clause: String = "SELECT")
```

Comment:
now the name `PipeSelect` becomes a bit weird. How about just `case class PipeExpression(..., isAgg: Boolean, clause: String)`

Reply:
`isAgg = true` means the expression must contain aggregate function, false means must not contain agg func

Reply:
Sure, sounds good, this is done.